util/grav_comp.hpp

```diff
@@ -1,66 +1,56 @@
 #pragma once
+
 #include "robot_model.hpp"
 #include "group_feedback.hpp"
 #include "Eigen/Dense"
 
 namespace hebi {
 namespace util {
 
-  static Eigen::VectorXd getGravityCompensationEfforts(
-    const hebi::robot_model::RobotModel& model,
-    const Eigen::VectorXd& masses,
-    const Eigen::VectorXd& positions,
-    const Eigen::Vector3d& gravity
-)
-  {
-
-    // Normalize gravity vector (to 1g, or 9.8 m/s^2)
-    Eigen::Vector3d normed_gravity = gravity;
-    normed_gravity /= normed_gravity.norm();
-    normed_gravity *= 9.81;
-
-    size_t num_dof = model.getDoFCount();
-    size_t num_frames = model.getFrameCount(HebiFrameTypeCenterOfMass);
-
-    hebi::robot_model::MatrixXdVector jacobians;
-    model.getJ(HebiFrameTypeCenterOfMass, positions, jacobians);
-
-    // Get torque for each module
-    // comp_torque = J' * wrench_vector
-    // (for each frame, sum this quantity)
-    Eigen::VectorXd comp_torque(num_dof);
-    comp_torque.setZero();
-
-    // Wrench vector
-    Eigen::VectorXd wrench_vec(6); // For a single frame; this is (Fx/y/z, tau x/y/z)
-    wrench_vec.setZero();
-    for (size_t i = 0; i < num_frames; ++i)
-    {
-      // Set translational part
-      for (size_t j = 0; j < 3; ++j)
-      {
-        wrench_vec[j] = -normed_gravity[j] * masses[i];
-      }
-
-      // Add the torques for each joint to support the mass at this frame
-      comp_torque += jacobians[i].transpose() * wrench_vec;
+inline Eigen::VectorXd getGravityCompensationEfforts(const hebi::robot_model::RobotModel& model, 
```

Comment:
`inline` modifier still makes the function "publicly" available (default visibility as opposed to static linkage) in each translation unit, but marking it as `inline` specifies to the linker that only one implementation needs to be chosen at link time. So technically, if you have two different implementations of function `foo`, which are both marked inline, it's not defined which version will be linked in. but you're doing weird stuff to begin with, if you do that...